ruby/lib/minitest/queue.rb

```diff
@@ -168,6 +181,163 @@ def flaky?
       end
     end
 
+    class PipeQueue
+      def initialize(deliminiter: nil)
+        @in, @out = IO.pipe
+        @deliminiter = deliminiter
+      end
+
+      def push(object)
+        payload = Marshal.dump(object)
+        out.write("#{payload}#{deliminiter}")
```

Comment:
Rather than a delimiter, it might make more sense to prefix the payload with it's size, e.g:

`[payload.bytesize].pack('N')`.

This way the child can read the header that is a known size, interpret it and then know exactly how much more bytes it needs to read.

Reply:
To clarify, I mean to to write `<payload-size><payload>`. This way you can keep a single pipe, rather than one for lenght and one for payloads.